drivers/adc/adc_stm32.c

```diff
@@ -1347,6 +1348,117 @@ static int adc_stm32_channel_setup(const struct device *dev,
 	return 0;
 }
 
+#if defined(CONFIG_SOC_SERIES_STM32C0X) || defined(CONFIG_SOC_SERIES_STM32G0X) ||                  \
+	defined(CONFIG_SOC_SERIES_STM32L0X) ||                                                     \
+	(defined(CONFIG_SOC_SERIES_STM32WBX) && defined(ADC_SUPPORT_2_5_MSPS)) ||                  \
+	defined(CONFIG_SOC_SERIES_STM32WLX) || defined(CONFIG_SOC_SERIES_STM32H7X)
+static bool adc_stm32_is_clk_sync(const struct adc_stm32_cfg *config)
+{
+	if (config->clk_prescaler == LL_ADC_CLOCK_SYNC_PCLK_DIV1 ||
+	    config->clk_prescaler == LL_ADC_CLOCK_SYNC_PCLK_DIV2 ||
+	    config->clk_prescaler == LL_ADC_CLOCK_SYNC_PCLK_DIV4) {
+		return true;
+	}
+
+	return false;
+}
+#endif
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+static inline int adc_stm32_get_input_freq_prescaler(void)
+{
+	int presc = 2;
+
+	/* For revision Y we have no prescaler of 2 */
+	if (LL_DBGMCU_GetRevisionID() <= 0x1003) {
+		presc = 1;
+	}
```

Comment:
```suggestion
#ifdef ADC_VER_V5_X
	/* For revision Y we have no prescaler of 2 */
	if (LL_DBGMCU_GetRevisionID() <= 0x1003) {
		presc = 1;
	}
#endif
```

This specificity is only present on STM32H742, H743, H750 and H753. So add an #ifdef to avoid conflict with other subseries (like H72x or H73x whose revisions are 0x1000 or 0x1001)